aws/src/test/java/org/apache/iceberg/aws/glue/TestGlueCatalog.java

```diff
@@ -0,0 +1,260 @@
+import software.amazon.awssdk.services.glue.model.UpdateDatabaseRequest;
+import software.amazon.awssdk.services.glue.model.UpdateDatabaseResponse;
+
+public class TestGlueCatalog {
+
+  private static final String WAREHOUSE_PATH = "s3a://bucket";
+  private GlueClient glue;
+  private GlueCatalog glueCatalog;
+
+  @Before
+  public void before() {
+    glue = Mockito.mock(GlueClient.class);
+    glueCatalog = new GlueCatalog(new Configuration(), glue, null, WAREHOUSE_PATH, false);
+  }
+
+  @Test
+  public void defaultWarehouseLocation_noDbUri() {
+    Mockito.doReturn(GetDatabaseResponse.builder()
+        .database(Database.builder().name("db").build()).build())
+        .when(glue).getDatabase(Mockito.any(GetDatabaseRequest.class));
+    String location = glueCatalog.defaultWarehouseLocation(TableIdentifier.of("db", "table"));
+    Assert.assertEquals(WAREHOUSE_PATH + "/db.db/table", location);
+  }
+
+  @Test
+  public void defaultWarehouseLocation_dbUri() {
+    Mockito.doReturn(GetDatabaseResponse.builder()
+        .database(Database.builder().name("db").locationUri("s3://bucket2/db").build()).build())
+        .when(glue).getDatabase(Mockito.any(GetDatabaseRequest.class));
+    String location = glueCatalog.defaultWarehouseLocation(TableIdentifier.of("db", "table"));
+    Assert.assertEquals("s3://bucket2/db/table", location);
+  }
+
+  @Test
+  public void listTables() {
+    Mockito.doReturn(GetDatabaseResponse.builder()
+        .database(Database.builder().name("db1").build()).build())
+        .when(glue).getDatabase(Mockito.any(GetDatabaseRequest.class));
+    Mockito.doReturn(GetTablesResponse.builder()
+        .tableList(
+            Table.builder().databaseName("db1").name("t1").build(),
+            Table.builder().databaseName("db1").name("t2").build()
+        ).build())
+        .when(glue).getTables(Mockito.any(GetTablesRequest.class));
+    Assert.assertEquals(
+        Lists.newArrayList(
+            TableIdentifier.of("db1", "t1"),
+            TableIdentifier.of("db1", "t2")
+        ),
+        glueCatalog.listTables(Namespace.of("db1"))
+    );
+  }
+
+  @Test
+  public void dropTable() {
+    Map<String, String> properties = new HashMap<>();
+    properties.put(BaseMetastoreTableOperations.TABLE_TYPE_PROP,
+        BaseMetastoreTableOperations.ICEBERG_TABLE_TYPE_VALUE);
+    Mockito.doReturn(GetTableResponse.builder()
+        .table(Table.builder().databaseName("db1").name("t1").parameters(properties).build()).build())
+        .when(glue).getTable(Mockito.any(GetTableRequest.class));
+    Mockito.doReturn(GetDatabaseResponse.builder()
+        .database(Database.builder().name("db1").build()).build())
+        .when(glue).getDatabase(Mockito.any(GetDatabaseRequest.class));
+    Mockito.doReturn(DeleteTableResponse.builder().build())
+        .when(glue).deleteTable(Mockito.any(DeleteTableRequest.class));
+    glueCatalog.dropTable(TableIdentifier.of("db1", "t1"));
```

Comment:
This doesn't appear to verify that any of the above calls actually happened. Mock tests should validate that the expected RPC calls were actually made.